churchland_pipeline_python/equipment.py

```diff
@@ -37,44 +50,310 @@ class Parameter(dj.Lookup):
+    electrode_tip_profile     = 'linear': enum('linear','curved','sharp') # tip profile
+    electrode_tip_insulation  = 0:        decimal(9,9) unsigned           # (m) insulation coverage, starting from tip bottom
+    """
+
+    contents = [
+        #id  |base-x   |base-y   |base-z   |base shape |base ins. |base rot. |tip-z   |tip prof. |tip ins.
+        [0,   12e-6,    12e-6,    0,        'cuboid',   0,         0,         0,       'linear',  0],        # flat square (e.g., Neuropixels)
+        [1,   15e-6,    15e-6,    0,        'cylinder', 0,         0,         0,       'linear',  0],        # flat circle (e.g., S-Probes)
+        [2,   100e-6,   100e-6,   0,        'cylinder', 0,         0,         1.5e-3,  'sharp',   0],        # cone (e.g., Utah array)
+        [3,   100e-6,   100e-6,   139.5e-3, 'cylinder', 129.5e-3,  0,         0.5e-3,  'sharp',   0.4e-3],   # sharp cylinder w/ insulation (e.g., FHC sharp electrode)
+        [4,   50e-6,    50e-6,    123e-3,   'cuboid',   0,         0,         2e-3,    'linear',  0],        # blunt cylinder w/ insulation (e.g., Natus hook-wire stock)
+        [5,   50e-6,    50e-6,    120e-3,   'cuboid',   0,         0,         5e-3,    'linear',  3e-3],     # blunt cylinder w/ insulation (e.g., Natus hook-wire stock)
+    ]
+
+    def plot(self, center_coords, resolution=100):
+
+        assert len(self) == 1, 'Specify one entry for plotting'
+
+        # fetch electrode parameters
+        params = self.fetch1()
+
+        # ---------
+        # PLOT BASE
+        # ---------
+
+        # base axes lengths
+        base_axes_lengths = np.array([v for k,v in params.items() if re.search(r'base_\w_length',k) is not None])
+
+        fig = plt.figure()
+
+        # plot 2D
+        if params['electrode_base_z_length'] == params['electrode_tip_z_length'] == 0:
+
+            # base coordinates
+            if params['electrode_base_shape'] == 'cuboid': # rectangle
+
+                base_coords = (base_axes_lengths * np.array([
+                    [ 0.5,-0.5,0],
+                    [ 0.5, 0.5,0],
+                    [-0.5, 0.5,0],
+                    [-0.5,-0.5,0]])).T
+                base_coords = np.concatenate((base_coords, base_coords[:,0,None]), axis=1)
+
+            else: # ellipse
+
+                theta = np.linspace(0, 2*np.pi, resolution)
+                base_coords = np.concatenate((\
+                    base_axes_lengths[0]/2 * np.cos(theta)[:,np.newaxis],\
+                    base_axes_lengths[1]/2 * np.sin(theta)[:,np.newaxis],\
+                    np.zeros((len(theta),1))), axis=1).T
+
+            # plot base at each center
+            for center in center_coords:
+                plt.plot(center[0]+base_coords[0,:], center[1]+base_coords[1,:], 'k')
+
+        else: # plot 3D
+
+            ax = fig.gca(projection='3d')
+
+            if params['electrode_base_shape'] == 'cuboid': # cuboid
+
+                plot_shape = 'cuboid'
```

Comment:
not used

Reply:
Indeed, this is still under development. I suppose this would have been a good time to use branches...will remove this function from the master branch in the next update